src/script_opt/ZAM/OPs/unary-exprs.op

```diff
@@ -164,14 +164,6 @@ macro EvalCast(lhs, rhs)
 	else
 		ERROR(error.c_str());
 
-# Cast an "any" type to the given type.  Only needed for type-based switch
```

Comment:
1. To know that it's unused: (a) `grep -r` in `src/` on the operand's name to see whether it's directly used. (b) `grep build/src/ZAM*` to see whether it's indirectly used (by some code generated by `gen-zam`). Note that here there will be hits, but if they all relate to defining the operand and not actually using it, then the operand isn't being used. (c) Remove it and compile - the ultimate test - since if things compile fine, then the operand is definitely not used.
2. Removing it doesn't fix a leak, that's a cross-talk commit-message error I introduced when making a change just a few lines above (that turned out to be wrong, and is removed in a `fixup`). I'll aim to remedy that when rebasing.

Reply:
Ah cool — fwiw, I grepped a bunch, and below `build`, but I frequently landed no hits, so I wondered how this one is different. I might just be looking for the wrong thing — would it be `Cast-Any` in this case? I noticed that the names seem to get mangled, as in `Field-LHS-Assign` -> `Field_LHS_Assign*`, `FieldLHSAssignExp` etc.

Reply:
Yeah there are two types of name groupings like that. One relates to C++ methods, like the example in your final comment, which aren't germane here. The second concerns ZAM instruction operands. For these, the starting point for this instance would be:
```
cd build
grep -i cast.any src/ZAM*
```
to look for any operands with names derived from the template's name (`Cast-Any`). The alphanumerics in names are preserved, but casing and `-`s are not. In this case, that turns up:
```
src/ZAM-EvalDefs.h:case OP_CAST_ANY_VV:
src/ZAM-Op1FlavorsDefs.h:	OP1_WRITE,	// OP_CAST_ANY_VV
src/ZAM-OpSideEffects.h:	false,	// OP_CAST_ANY_VV
src/ZAM-OpsDefs.h:	OP_CAST_ANY_VV,
src/ZAM-OpsNamesDefs.h:	case OP_CAST_ANY_VV:	return "cast-any-VV";
```
which is enough to see that `gen-zam` created each of those, but didn't create any C++ code that _uses_ those. You'd then try
```
cd ../src
grep -r OP_CAST_ANY
```
to see if the compiler uses the instruction directly. If not, then it can be removed.

I'll add this example to the internals doc.

Reply:
Yeah there are two types of name groupings like that. One relates to C++ methods, like the example in your final comment, which aren't germane here. The second concerns ZAM instruction operands. For these, the starting point for this instance would be:
```
cd build
grep -i cast.any src/ZAM*
```
to look for any operands with names derived from the template's name (`Cast-Any`). The alphanumerics in names are preserved, but casing and `-`s are not. In this case, that turns up:
```
src/ZAM-EvalDefs.h:case OP_CAST_ANY_VV:
src/ZAM-Op1FlavorsDefs.h:	OP1_WRITE,	// OP_CAST_ANY_VV
src/ZAM-OpSideEffects.h:	false,	// OP_CAST_ANY_VV
src/ZAM-OpsDefs.h:	OP_CAST_ANY_VV,
src/ZAM-OpsNamesDefs.h:	case OP_CAST_ANY_VV:	return "cast-any-VV";
```
which is enough to see that `gen-zam` created each of those, but didn't create any C++ code that _uses_ those. You'd then try
```
cd ../src
grep -r OP_CAST_ANY
```
to see if the compiler uses the instruction directly. If not, then it can be removed.

I'll add this example to the internals doc.